libraries/stdlib/js/src/js/rtti.js

```diff
@@ -39,22 +39,35 @@ Kotlin.callSetter = function (thisObject, klass, propertyName, value) {
     Kotlin.callSetter(thisObject, Object.getPrototypeOf(klass), propertyName, value);
 };
 
-function isInheritanceFromInterface(ctor, iface) {
-    if (ctor === iface) return true;
-
+function getAllInterfaces(ctor) {
+    if (ctor.$metadata$ == null) ctor.$metadata$ = {};
     var metadata = ctor.$metadata$;
-    if (metadata != null) {
+    if (metadata._allInterfaces === undefined) {
+        var allInterfaces = metadata._allInterfaces = [];
+
+        allInterfaces.push(ctor);
```

Comment:
actually, it's not all interfaces, but all parents + class itself

Reply:
I think we can push `superConstructor` at 60 line instead of pushing `ctor` here.